internal/cli/cmd/plugin/build/build.go

```diff
@@ -0,0 +1,143 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/mholt/archives"
+	"github.com/perses/perses/internal/api/archive"
+	"github.com/perses/perses/internal/api/plugin"
+	persesCMD "github.com/perses/perses/internal/cli/cmd"
+	"github.com/perses/perses/internal/cli/output"
+	"github.com/spf13/cobra"
+)
+
+type option struct {
+	persesCMD.Option
+	skipNPMBuild   bool
+	archiveFormat  archive.Format
+	frontendFolder string
+	schemaFolder   string
+	distFolder     string
+	writer         io.Writer
+	errWriter      io.Writer
+}
+
+func (o *option) Complete(args []string) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no args are supported by the command 'update'")
+	}
+	return nil
+}
+
+func (o *option) Validate() error {
+	if !archive.IsValidFormat(o.archiveFormat) {
+		return fmt.Errorf("archive format %q not managed", o.archiveFormat)
+	}
+	return nil
+}
+
+func (o *option) Execute() error {
+	// First step: run npm to build the frontend part.
+	if !o.skipNPMBuild {
+		if err := exec.Command("npm", "run", "build").Run(); err != nil {
+			return fmt.Errorf("unable to build the frontend: %w", err)
+		}
+	}
+	// Get the plugin name from the manifest file
+	manifest, err := plugin.ReadManifest(o.distFolder)
+	if err != nil {
+		return fmt.Errorf("unable to read manifest: %w", err)
+	}
+
+	// Then we need to create the archive.
+	// The archive contains the following file:
+	// - package.json: required to get the type of the plugin and the name
+	// - mf-manifest.json and mf-stats.json: contains the plugin name
+	// - static: folder containing the UI part
+	// - schemas: folder containing the schema files
+	files, err := o.computeArchiveFiles()
+	if err != nil {
+		return err
+	}
+	if archiveBuildErr := archive.Build(manifest.Name, o.archiveFormat, files); archiveBuildErr != nil {
+		return fmt.Errorf("archive creation failed: %w", archiveBuildErr)
+	}
+	return output.HandleString(o.writer, fmt.Sprintf("Building %s successfully", manifest.Name))
+}
+
+func (o *option) computeArchiveFiles() ([]archives.FileInfo, error) {
+	list := make(map[string]string)
+	// add README and LICENSE if they are present as they are optional
+	const readme = "README.md"
+	if _, err := os.Stat(readme); err == nil {
+		list[readme] = readme
+	}
+	const license = "LICENSE"
+	if _, err := os.Stat(license); err == nil {
+		list[license] = license
+	}
+	// Add the dist content at the root of the archive (saying differently, dist folder should not appear in the archive)
+	distFiles, err := os.ReadDir(o.distFolder)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read 'dist' directory: %w", err)
+	}
+	for _, f := range distFiles {
+		list[path.Join(o.distFolder, f.Name())] = f.Name()
+	}
+
+	// Do the same for the Cuelang schemas
```

Comment:
We also need to embed the `cue.mod` folder here.

And (maybe for a followup PR?) I guess we should run `percli plugin update` before that in order to ensure that `cue.mod/pkg` is populated with the common deps.

Reply:
I was thinking that as well and then I'm thinking that perhaps in some case, you don't necessary need to get the cue dependencies. For example the markdown panel doesn't depend on anything and as such doesn't require to run `percli plugin update`